recipe-server/normandy/base/api/views.py

```diff
@@ -0,0 +1,43 @@
+from urllib.parse import urljoin
+
+from django.conf import settings
+
+from rest_framework.compat import NoReverseMatch
+from rest_framework.response import Response
+from rest_framework.reverse import reverse
+from rest_framework.views import APIView
+
+
+class APIRootView(APIView):
+    """
+    An API root view that lists the urls passed to it via api_urls.
+    """
+    _ignore_model_permissions = True
+    exclude_from_schema = True
+    api_urls = []
+
+    def get(self, request, *args, **kwargs):
+        ret = {}
+
+        namespace = getattr(request.resolver_match, 'namespace', None)
+        for api_url in self.api_urls:
+            if api_url.name is None:
+                continue
+            url_name = api_url.name
+            if namespace:
+                url_name = namespace + ':' + url_name
+
+            try:
+                view_class = getattr(api_url.callback, 'view_class', None)
+                if getattr(view_class, 'always_dynamic', False) and settings.APP_SERVER_URL:
```

Comment:
A flag on a class seems awkward because it only affects this listing, and not anywhere we reverse a URL to the class. I dunno if it's possible, but a better API IMO would be adding a parameter to `router.register` and `router.register_view` called `allow_cdn` or something to control this.